src/Engine/ProtoCore/Utils/ArrayUtils.cs

```diff
@@ -117,18 +117,25 @@ public static ClassNode GetGreatestCommonSubclassForArray(StackValue array, Runt
             return runtimeCore.DSExecutable.classTable.ClassNodes[orderedTypes.First()];
         }
 
-        public static Dictionary<int, StackValue> GetTypeExamplesForLayer(StackValue array, RuntimeCore runtimeCore)
+        /// <summary>
+        /// This method returns the distinct reduced params for all the elements inside the
```

Comment:
I think we should be clear about what `distinct` means here. In this case it is different types, so this method gives you one `example` per type from type from the first level of the array.